src/main/java/com/atharvadholakia/password_manager/controller/CredentialController.java

```diff
@@ -34,6 +34,29 @@ public ResponseEntity<Credential> createCredential(@Valid @RequestBody Credentia
     return new ResponseEntity<>(createdCredential, HttpStatus.CREATED);
   }
 
+  @PatchMapping("/{id}")
+  public ResponseEntity<Credential> updateCredential(
+      @PathVariable String id, @Valid @RequestBody Credential credential) {
+    Credential oldCredential = credentialService.getCredentialById(id);
```

Comment:
Incase, the provided `id` is incorrect and `oldCredential` doesn't exist. Is this case handled?

Reply:
Yes, It is taken care of. It will throw 404 ResourceNotFoundException. I have also written a testcase for the same.